### ISSUE TITLE
AL.exe should be converted to .Net Core

### ISSUE THREAD

**Issue by joperezr** [Microsoft/msbuild#350]
In order to build corefx repo using .Net Core version of msbuild, we require al.exe to run in .NetCore since we need to generate satellite assemblies. Today, if you try to build a project that needs this task will fail when trying to load it. There should be a workaround for the time being to disable the task from running, and once Al.exe is converted to net core it should be re-added back.

For this, I've used the .NetCore msbuild produced by https://github.com/dsplaisted/msbuild/tree/netcore-selfbuild

**Comment by joperezr** [comment]
cc @dsplaisted

**Comment by dsplaisted** [comment]
It looks like this was needed for localization.  @weshaggard, what are your plans around enabling localization?  Will you need an equivalent to AL.exe on NET Core?

**Comment by weshaggard** [comment]
At this point we don't plan to ship satellite assemblies for RTM but we will eventually need support for producing them.

**Comment by nguerrera** [comment]
@jaredpar

**Comment by rainersigwald** [comment]
That seems reasonable to me, and 16.0/3.0 like a pretty good time for it, if we can fit it in.

**Comment by livarcocc** [comment]
I don't believe this issue belongs in the msbuild repo given that AL.exe is not really owned by us. Now that we have dotnet/winforms, moving the issue there, I think that's the right repo.

**Comment by livarcocc** [comment]
This issue was moved to dotnet/winforms#586

**Comment by nguerrera** [comment]
AL is owned by @jaredpar's team, not winforms.

**Comment by nguerrera** [comment]
Nothing is ever getting fixed in AL. We have the power to do what I suggested above in https://github.com/Microsoft/msbuild/issues/350#issuecomment-433533045

**Comment by NightOwl888** [comment]
What options are available for doing satellite assembly generation cross platform? The [Create satellite assemblies for .NET apps](https://docs.microsoft.com/en-us/dotnet/core/extensions/create-satellite-assemblies) documentation steers us toward using al.exe.

I am creating a dotnet tool that can be used to compile satellite assemblies. Ideally, there would be a NuGet package I can import to generate the assemblies similar to the command:

```c#
al -target:lib -culture:de -out:Example.resources.dll -template:Example.dll -embed:de.res  -embed:feature1.de.res -embed:feature2.de.res -keyfile:key.snk
```

It is not particularly complicated, we are not even delay signing.

The title of this issue indicates to switch to using csc.exe. Is this included in the .NET 6 SDK? If so, where is it be found?

We don't have an issue with creating a `.csproj` file in order to use dotnet build and `new Process()` to pull this off (since we depend on the SDK anyway), I am just trying to figure out the right angle of attack to approach the issue.
